src/Bicep.Local.Deploy/Extensibility/LocalExtensibilityHandler.cs

```diff
@@ -5,61 +5,70 @@
 
-    public LocalExtensibilityHandler(IModuleDispatcher moduleDispatcher, Func<Uri, Task<LocalExtensibilityProvider>> providerFactory)
+    public LocalExtensibilityHandler(IModuleDispatcher moduleDispatcher, Func<Uri, Task<LocalExtensibilityExtension>> extensionFactory)
     {
         this.moduleDispatcher = moduleDispatcher;
-        this.providerFactory = providerFactory;
+        this.extensionFactory = extensionFactory;
         // Built in provider for handling nested deployments
-        RegisteredProviders[new("LocalNested", "0.0.0")] = new AzExtensibilityProvider(this);
+        RegisteredExtensions[new("LocalNested", "0.0.0")] = new AzExtensibilityExtension(this);
     }
 
-    private async Task<ExtensibilityOperationResponse> CallProvider(string method, IExtensibilityProvider provider, ExtensibilityOperationRequest request, CancellationToken cancellationToken)
+    public async Task<ResourceResponseBody> CallExtensibilityHost(
+        ExtensionInfo extensionInfo,
+        HttpContent content,
+        CancellationToken cancellationToken)
     {
-        return method switch
-        {
-            "get" => await provider.Get(request, cancellationToken),
-            "delete" => await provider.Delete(request, cancellationToken),
-            "save" => await provider.Save(request, cancellationToken),
-            "previewSave" => await provider.PreviewSave(request, cancellationToken),
-            _ => throw new NotImplementedException($"Unsupported method {method}"),
-        };
+        var extension = RegisteredExtensions[new(extensionInfo.ExtensionName, extensionInfo.ExtensionVersion)];
+
+        return await CallExtension(extensionInfo.Method, extension, content, cancellationToken);
     }
 
-    public async Task<ExtensibilityOperationResponse> CallExtensibilityHost(
+    private async Task<ResourceResponseBody> CallExtension(
         string method,
-        ExtensibilityOperationRequest request,
+        LocalExtensibilityExtension provider,
+        HttpContent content,
         CancellationToken cancellationToken)
     {
-        var provider = RegisteredProviders[new(request.Import.Provider, request.Import.Version)];
-
-        return await CallProvider(method, provider, request, cancellationToken);
+        return method switch
+        {
+            "get" => await provider.Get(await content.ReadAsAsync<ResourceReferenceRequestBody>(cancellationToken), cancellationToken),
+            "delete" => await provider.Delete(await content.ReadAsAsync<ResourceReferenceRequestBody >(cancellationToken), cancellationToken),
+            "createOrUpdate" => await provider.CreateOrUpdate(await content.ReadAsAsync<ResourceRequestBody>(cancellationToken), cancellationToken),
+            "preview" => await provider.Preview(await content.ReadAsAsync<ResourceRequestBody>(cancellationToken), cancellationToken),
+            _ => throw new NotImplementedException($"Unsupported method {method}"),
+        };
```

Comment:
It feels odd that this method deserializes the request (`content.ReadAsAsync<ResourceReferenceRequestBody>`) but then returns an unserialized response (`ResourceResponseBody`). Should we instead just have it return a `HttpContent` or equivalent?

Reply:
I would prefer using STJ for better performance. @jorgecotillo, we discussed using v2 models from [Azure.Deployments.Extensibility](https://www.nuget.org/packages/Azure.Deployments.Extensibility.Core/) instead. Could we switch to using those models?

Reply:
Done